### ISSUE TITLE
short form to declare a local variable with an initializer as nullable

### ISSUE THREAD

**Issue by sigmundch** [dart-lang/language#1238]
As people migrate to null-safety, I expect we'll find several cases like:
```dart
var x = "foo"; // x is String
...
x = null;
...
```

Currently as part of the migration they may need to do:
```dart
String? x = "foo";
...
```

Random thought - what if we allowed `var?` as a way to continue letting the system infer the type term, but impose the nullability explicitly:
```dart
var? x = "foo"; // x is `String?`
```

Thoughts?
/cc @munificent - this comes hand in hand with readability/style guidelines.

**Comment by eernstg** [comment]
Note that this is a near-duplicate of https://github.com/dart-lang/language/issues/1030 where it was noted that this change isn't planned, but makes sense and would be non-breaking.

**Comment by lrhn** [comment]
Yes, #1030 explained why the syntax isn't automatically valid (`var` is not a *type*), and said that it wasn't currently planned.

It's not a bad feature, just not one we're currently considering essential to launching null safety.

It *keeps* cropping up from new people, always with exactly the same syntax, which suggests that it's probably an easily explainable feature. 
It doesn't generalize to the other union type, `FutureOr<var> x = 42;` won't work (but if we fixed type inference for raw types, then `FutureOr x = 42;` might work instead.

**Comment by eernstg** [comment]
Why wouldn't `FutureOr<var> x = 42;` give `x` the inferred type `FutureOr<int>`? Of course, `FutureOr<var?>` is the next thing. ;-)
